programs/client/Client.h

```diff
@@ -29,6 +29,7 @@ class Client : public ClientBase
                         const std::vector<Arguments> & external_tables_arguments) override;
     void processConfig() override;
 
+    std::vector<String> hosts{};
```

Comment:
Where is a single host keeped? You need to find it and keep vector of hosts instead of one host

Reply:
A single host is kept originally in configuration object with type `Poco::Util:AbstactConfiguration`, which pointer you can get from method `config()`. Configuration is usually setting in method `Client::init`, and other methods, which are called in `init`.
To set field in `config()` you shoud call `config().set<String | Int | ...>(<key>, <value>)`. But there are setting methods only for simple types: int, string, bool, etc. You can't set key with `std::vector` value.

I watched, how vectors are saved in other situations (when other vector parametrs are set). I found param `--query-files` in class `ClientBase`: https://github.com/ManagedDatabases/ClickHouse/blob/managed-feature/choose-server-from-list/src/Client/ClientBase.cpp#L1536

And the values are saving in special field of class `ClientBase` (while other simple string or int params are saved in `config()`):  https://github.com/ManagedDatabases/ClickHouse/blob/managed-feature/choose-server-from-list/src/Client/ClientBase.cpp#L1600

That is why I've made an similar field for hosts in `Client` class.

Reply:
Well so you can add it to `ClientBase` as it is done for query-files